compiler/ml/ctype.ml

```diff
@@ -2762,12 +2762,14 @@ let filter_arrow env t l =
   let t = expand_head_trace env t in
   match t.desc with
   | Tvar _ ->
+    let _ = assert false in
+    (* TODO: need the arity from the function definition *)
```

Comment:
This needs some info from function definition that is not readily available. Better to postpone this until arity is explicit in function definitions.

Reply:
Here: https://github.com/rescript-lang/rescript/pull/7197